cmd/integration/integration.go

```diff
@@ -258,9 +258,15 @@ func makeTempDirOrDie(prefix string, baseDir string) string {
 }
 
 // podsOnMinions returns true when all of the selected pods exist on a minion.
-func podsOnMinions(c *client.Client, pods api.PodList) wait.ConditionFunc {
+func podsOnMinions(c *client.Client, podNamespace string, labelSelector labels.Selector) wait.ConditionFunc {
 	podInfo := fakeKubeletClient{}
+	// wait for minions to indicate they have info about the desired pods
 	return func() (bool, error) {
+		pods, err := c.Pods(podNamespace).List(labelSelector)
+		if err != nil {
+			glog.Infof("Unable to get pods to list: %v", err)
+			return false, nil
+		}
 		for i := range pods.Items {
 			host, id, namespace := pods.Items[i].Spec.Host, pods.Items[i].Name, pods.Items[i].Namespace
 			glog.Infof("Check whether pod %s.%s exists on node %q", id, namespace, host)
```

Comment:
Suggestion: is it enough to just look at status.phase == running and not use the kubelet client?